renderdoc/driver/d3d12/d3d12_pixelhistory.cpp

```diff
@@ -981,6 +1006,7 @@ struct D3D12ColorAndStencilCallback : public D3D12PixelHistoryCallback
 
     size_t storeOffset =
         m_EventIndices.size() * sizeof(D3D12EventInfo) + offsetof(struct D3D12EventInfo, premod);
+    m_SavedState = m_pDevice->GetQueue()->GetCommandData()->GetCurRenderState();
```

Comment:
I am not sure the need for this line here. `CopyPixel()` should only be reading from `m_SavedState`. Any modifications to the current render state should be being saved into `m_SavedState` before modifications and restored after modifications.

If you found a situation where `m_SavedState` was incorrect and this line fixed it then be good to work backwards to see where `m_SavedState` did not get restored.

Reply:
This issue is not a validation problem directly, but I found it via an indirectly resulting validation error being thrown. The issue is that CopyPixel uses the value of m_SavedState for the depth stencil, but when replaying a dispatch the value was not set. So uses of m_SavedState.dsv appeared to be reading uninitialized data, in this case causing the srcImageState to be set to D3D12_RESOURCE_STATE_DEPTH_READ which causes an invalid resource barrier to be used. 
In retrospect it is unclear to me why the memory for this member appeared to be filled with garbage, when I would expect a class member to receive default initialization. I'll have to double check that. 
I think also from a code style standpoint perhaps the data being used from m_SavedState should be passed in as a parameter to avoid problems like this.